### PR TITLE
fix: DEV-1470: Tasks list page navigation

### PR DESCRIPTION
**Pull request by nicholasrq**

### CODE REVIEW

File: src/hooks/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useHistory } from 'react-router';
+import { useFixedLocation } from '../providers/RoutesProvider';
+
+export const useRefresh = () => {
+  const history = useHistory();
+  const { pathname } = useFixedLocation();
+
+  const refresh = useCallback((redirectPath) => {
+    history.replace("/");
```
Comment (hlomzik) [inline src/hooks/hooks.ts:10]: add comments

File: src/hooks/useFirstMountState.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { useRef } from "react";
+
+export function useFirstMountState(): boolean {
```
Comment (hlomzik) [inline src/hooks/useFirstMountState.ts:3]: dublicate

File: src/hooks/useUpdateEffect.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { useEffect } from "react";
+import { useFirstMountState } from "./useFirstMountState";
+
+export const useUpdateEffect: typeof useEffect = (effect, deps) => {
```
Comment (hlomzik) [inline src/hooks/useUpdateEffect.ts:4]: dublicate

File: src/components/Common/Button/Button.js [hunk 4]
```diff
@@ -45,13 +45,15 @@ export const Button = forwardRef(
       }
     }, [icon, size]);
 
+    console.log({ className: className?.toString() });
```
Comment (hlomzik) [inline src/components/Common/Button/Button.js:48]: remove

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: src/hooks/useUpdateEffect.ts (hunk 3 above)
Comment (nicholasrq) [reply src/hooks/useUpdateEffect.ts:4]: `hooks.ts` was removed in favor of separate hook-per-file

File: src/hooks/useFirstMountState.ts (hunk 2 above)
Comment (nicholasrq) [reply src/hooks/useFirstMountState.ts:3]: `hooks.ts` was removed in favor of separate hook-per-file

File: src/hooks/hooks.ts (hunk 1 above)
Comment (nicholasrq) [reply src/hooks/hooks.ts]: removed. was accidentally added from the LSP, not needed in DM

File: src/components/Table/Table.js [hunk 5]
```diff
@@ -55,14 +56,21 @@ export const DataView = injector(
...
+    const setPage = useCallback((page, pageSize) => {
+      setPageSize(pageSize);
+      setStoredPageSize("tasks", pageSize);
+    }, []);
+
     const focusedItem = useMemo(() => {
       return props.focusedItem;
     }, [props.focusedItem]);
 
     const loadMore = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (nicholasrq) [reply src/components/Table/Table.js]: removed

**Review by hlomzik — APPROVED** [review]

**Pull request merged**